sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -383,14 +383,20 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
         // Get all input data source relations of the query.
         val srcRelations = df.logicalPlan.collect {
           case LogicalRelation(src: BaseRelation, _, _) => src
+          case relation: CatalogRelation if DDLUtils.isHiveTable(relation.catalogTable) =>
+            relation.catalogTable.qualifiedName
```

Comment:
why not use `identifier`? I think it's more concise to represent table identifier.